src/client/components/ModalNew.tsx

```diff
@@ -115,14 +106,7 @@ export const Modal: React.FC<ModalProps> = ({
   onClose,
   children,
 }) => {
-  React.useEffect(() => {
-    if (isVisible) {
-      document.body.style.overflow = 'hidden'
-    }
-    return () => {
-      document.body.style.overflow = 'visible'
-    }
-  }, [isVisible])
+  useLockBodyScroll({ lock: isVisible })
```

Comment:
Like I mentioned to @gustaveen, I think it's a nice touch to name custom hooks that case side effects like:

`useLockBodyScrollEffect`

Just to make it clear without having to look at the implementation.

Reply:
🤔 Dunno about that. I mean, adding 'effect' in the name kind of suggest a pattern where we "import" how something was implemented (implementation details) to its name, which I think it's not usually a good thing.

Or maybe I misunderstood your point and your suggestion is based on a concern of yours that is generally good for a consumer component to know that a custom hook is causing a side effect?! Is that the case?

Reply:
Yes, exactly! Just to give an indication to the one reading the code that when this hook is called some side effect will happen :)

Reply:
🤔 I see. The only "problem" is it feels we would need to think when add or not the "effect" word in the name. Take `useQuoteIds` for example. It would feel odd if its name was `useQuoteIdsEffect`, right?! What do you think?

Reply:
Aha good point - I agree that the name of that hook is not optimal.

I would name it something like `usePersistedQuoteIds`.

If it's a hook that _only_ does a side effect though I think it's still pretty straight forward. But we should write all these conventions down so totally fine to leave as is for this PR.